### PR TITLE
Add a jsonp dependency to quarkus-oidc

### LINKED ISSUE #4672: OIDC Extension throws ClassNotFoundException when using quarkus-resteasy-jackson
**Describe the bug**
I am migration a project with the Keycloak extension (Quarkus 0.22.0) to the new OIDC extension (Quarkus 0.25.0). I am using the `quarkus-resteasy-jackson` extension instead of jsonb. Unfortunately, these extensions are no longer interchangeable: the access token cannot be parsed, because org.eclipse.microprofile.jwt.Claims specifies JsonObject types, e.g. for the address.
See [stacktrace_JsonObject_notFound.txt](https://github.com/quarkusio/quarkus/files/3744710/stacktrace_JsonObject_notFound.txt)

**Expected behavior**
The app does not crash. Required dependencies are provided.

**Actual behavior**
ClassNotFoundException

**To Reproduce**
1. In the `using-openid-connect` starter pom, change the dependency `quarkus-resteasy-jsonb` to `quarkus-resteasy-jackson` 
2. access UserResource as described in the [guide](https://quarkus.io/guides/oidc-guide.html).

### PR DESCRIPTION
**Pull request by sberyozkin**
Fixes #4672

### CODE REVIEW

**Comment by sberyozkin** [comment]
The #4672 issue is that `quarkus-oidc` does not include `quarkus-jsonp` as it currently depends directly on `smallrye-jwt` as opposed to `quarkus-smallrye-jwt` which does include `quarkus-jsonp`.

**Comment by geoand** [comment]
@sberzyokin can you elaborate on this a little?

**Comment by sberyozkin** [comment]
@geoand Just typed it :-)

**Comment by gsmet** [comment]
So it's not supposed to depend on `quarkus-smallrye-jwt`?

**Comment by gsmet** [comment]
Looks like our JSON-P extension is missing something:
```
Caused by: java.lang.InstantiationException: Type `org.glassfish.json.JsonProviderImpl` can not be instantiated reflectively as it does not have a no-parameter constructor or the no-parameter constructor has not been added explicitly to the native image.
	at java.lang.Class.newInstance(DynamicHub.java:770)
	at javax.json.spi.JsonProvider.provider(JsonProvider.java:97)
	... 71 more
```

File: extensions/oidc/runtime/pom.xml [hunk 1]
```diff
@@ -30,6 +30,10 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-security</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-jsonp</artifactId>
+        </dependency>
```
Comment (gsmet) [inline extensions/oidc/runtime/pom.xml]: I don't know if the right fix is to add this one or to depend on `quarkus-smallrye-jwt`. I let you judge of that, not knowing exactly what the SmallRye JWT extension is doing (but my fear is that we might miss some important native image-related stuff somehow).

In any case, you also need to add the `-deployment` dependency to the deployment module. That's why the tests are failing right now.

**Review by gsmet — CHANGES_REQUESTED** [review]

File: extensions/oidc/runtime/pom.xml (hunk 1 above)
Comment (sberyozkin) [reply extensions/oidc/runtime/pom.xml]: @gsmet sorry, I was working in replacing the test code with the Jackson provider and it worked, but not sure why I forgot to recompile all of the `oidc` extension...Let me answer about `quarkus-smallrye-jwt` in a few mins

**Comment by sberyozkin** [comment]
@geoand, @gsmet Hi, sorry for a delay... So there is #3064. Stuart did the initial integration directly with `smallrye-jwt` to support the `OIDC token to be accessed via MP-JWT API` case as at the moment we don't have a mechanism for `quarkus-smallrye-jwt` to complement the `primary` authentication mechanism offered in this case by `quarkus-oidc` while disabling its own authentication mechanism and due to this issue it is not obvious yet if this dependency has to be introduced after all because it may be simpler for `quarkus-oidc` to work directly with `smallrye-jwt`, not sure yet :-)

**Comment by geoand** [comment]
Thanks for the explanation! Although I have to admit that I haven't followed the oidc stuff so I can't really provide any meaningful feedback here...
If you are confident that this is the proper fix and if @gsmet doesn't have any objections, I propose we merge it.

**Comment by sberyozkin** [comment]
Hi Georgios, thanks, yes, at this moment of time this is the right fix

**Comment by sberyozkin** [comment]
Well, a bit more work to be done, but it is purely a test issue now, sorry

**Comment by geoand** [comment]
Cool, np!

**Comment by sberyozkin** [comment]
@geoand I'm a bit all over the place here at home tight now :-), I've done a quick native test fix, the real point of the PR is to prove that Jackson is not interfering into the OIDC adapter work, and as far as a complete JSONP to String/List/Map is concerned - we can deal with it once the test starts growing to check more complex claims...

**Comment by sberyozkin** [comment]
By the way, not sure why Jackson only started complaining about failing to convert a JSONString only in the native mode, I forgot earlier to add `-Dnative` and it all worked OK...

**Comment by geoand** [comment]
I'm cool with it 😎. Let's see if @gsmet agrees.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Merged, thanks.

**Pull request merged**